pkg/controllers/job/job_controller_actions.go

```diff
@@ -723,6 +723,15 @@ func (cc *jobcontroller) createOrUpdatePodGroup(job *batch.Job) error {
 				PriorityClassName: job.Spec.PriorityClassName,
 			},
 		}
+		if job.Spec.NetworkTopology != nil {
+			nt := &scheduling.NetworkTopologySpec{
+				Mode: scheduling.NetworkTopologyMode(job.Spec.NetworkTopology.Mode),
```

Comment:
Do we need to check whether `Mode` is nil?

Reply:
Mode is a string type and no panic problem, also the API schema has the constrain that it must be an enum value `soft` or `hard`.